apis/tests/v3/test_types.go

```diff
@@ -80,10 +80,12 @@ type Repository struct {
 	// commit id (sha) for checkout
 	Commit string `json:"commit,omitempty"`
 	// if needed we can checkout particular path (dir or file) in case of BIG/mono repositories
-	Path              string     `json:"path,omitempty"`
-	UsernameSecret    *SecretRef `json:"usernameSecret,omitempty"`
-	TokenSecret       *SecretRef `json:"tokenSecret,omitempty"`
-	CertificateSecret string     `json:"certificateSecret,omitempty"`
+	Path           string     `json:"path,omitempty"`
+	UsernameSecret *SecretRef `json:"usernameSecret,omitempty"`
+	TokenSecret    *SecretRef `json:"tokenSecret,omitempty"`
+	// if true git clone will be executed with the http.extraHeader arg set to the value specified in token
+	IsHeaderToken     *bool  `json:"isHeaderToken,omitempty"`
```

Comment:
this is right one

Reply:
why do you make it a pointer to bool?

Reply:
My idea is that when this value is set to "true" in the payload, the git clone command will be modified